pkg/apis/planetscale/v2/vitessshard_types.go

```diff
@@ -368,9 +368,9 @@ type VitessShardStatus struct {
 	// to turn down the shard.
 	Idle corev1.ConditionStatus `json:"idle,omitempty"`
 
-	// Conditions is a map of all VitessShard specific conditions we want to set and monitor.
+	// Conditions is a list of all VitessShard specific conditions we want to set and monitor.
 	// It's ok for multiple controllers to add conditions here, and those conditions will be preserved.
-	Conditions map[VitessShardConditionType]VitessShardCondition `json:"conditions,omitempty"`
+	Conditions []VitessShardCondition `json:"conditions,omitempty"`
```

Comment:
I think we should wait to change the existing conditions field on VitessShard. I'm worried that this change might break existing operator deployments upon upgrade due to validation/deserialization issues. Let's just make VitessKeyspace conditions start on the new format and we'll figure out the VitessShard migration separately so it doesn't block resharding work.

Reply:
Ok sounds good. I'll revert these changes then.

Reply:
Reverted.